Runtime/Code/Luau/LuauCoreCallbacks.cs

```diff
@@ -1439,8 +1436,6 @@ static unsafe int CallMethodCallback(LuauContext context, IntPtr thread, int ins
         catch (Exception e) {
             return LuauError(thread,
                 "Error: Exception thrown in method " + type.Name + "." + finalMethod.Name + ": " + e);
-        } finally {
-            Profiler.EndSample();

```

Comment:
If `finalMethod` happens to be `Profiler.BeginSample` being called from TS, then this profiler block breaks it. The logic goes:

1. CS: `Profiler.BeginSample("LuauCore.InvokeMethod");`
2. TS: `Profiler.BeginSample("foo");`
3. CS: `Profiler.EndSample();`

You can see that step 3 ends step 2, but its intention is to end step 1.

Because Liam already added profiling in the plugin for the full CallMethodCallback, this profile block doesn't seem super useful outside of simply profiling how long the actual method takes to invoke.